doc/source/user_guide/text.rst

```diff
@@ -265,35 +265,23 @@ i.e., from the end of the string to the beginning of the string:
        dtype="string",
    )
    s3
-   s3.str.replace("^.a|dog", "XX-XX ", case=False)
+   s3.str.replace("^.a|dog", "XX-XX ", case=False, regex=True)
 
-Some caution must be taken to keep regular expressions in mind! For example, the
-following code will cause trouble because of the regular expression meaning of
-``$``:
+Some caution must be taken when dealing with regular expressions! The current behavior
+is to treat single character patterns as literal strings, even when ``regex`` is set
+to ``True``. This behavior is deprecated and will be removed in a future version so
+that the ``regex`` keyword is always respected.
 
-.. ipython:: python
-
-   # Consider the following badly formatted financial data
-   dollars = pd.Series(["12", "-$10", "$10,000"], dtype="string")
-
-   # This does what you'd naively expect:
-   dollars.str.replace("$", "")
-
-   # But this doesn't:
-   dollars.str.replace("-$", "-")
+.. versionchanged:: 1.2.0
 
-   # We need to escape the special character (for >1 len patterns)
-   dollars.str.replace(r"-\$", "-")
-
-If you do want literal replacement of a string (equivalent to
-:meth:`str.replace`), you can set the optional ``regex`` parameter to
-``False``, rather than escaping each character. In this case both ``pat``
-and ``repl`` must be strings:
+If you want literal replacement of a string (equivalent to :meth:`str.replace`), you
+can set the optional ``regex`` parameter to ``False``, rather than escaping each
+character. In this case both ``pat`` and ``repl`` must be strings:
 
 .. ipython:: python
 
     # These lines are equivalent
-    dollars.str.replace(r"-\$", "-")
+    dollars.str.replace(r"-\$", "-", regex=True)
```

Comment:
can you comment on why here (e.g. a single pat is never treated as a regex)

Reply:
I think the equivalence here is because $ is being escaped (it isn't a single character regex)